### PR TITLE
Fix PHPStan issues and build on PHP 8.2

### PR DESCRIPTION
**Pull request by sanmai**
Looks like PHPStan is kinda lost here.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -24,11 +24,10 @@
  		"psr-4": { "Michelf\\": "Michelf/" }
 	},
 	"require-dev": {
-		"phpspec/prophecy": "^1.6",
```
Comment (sanmai) [inline composer.json:27]: There's no need for this dependency. See:
https://github.com/sebastianbergmann/phpunit/issues/5033#issuecomment-1221992857

**Comment by michelf** [comment]
I see you're hitting some problems with the PHPStan and the tests. I think it's confused by the implicit treatment of DOMNode as DOMText (and perhaps other similar things). I tried to fix it in #377, but ended up fixing the wrong things and abandoned because I had not time left for this.

Thank you for your work.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -26,4 +26,5 @@ jobs:
             -   run: composer install --no-progress
 #            -   run: composer codestyle
             -   run: composer phpstan
+                if:  matrix.php == '8.1'
```
Comment (sanmai) [inline .github/workflows/ci.yml:29]: If we run PHPStan on every version, we'll use different versions of PHPStan detecting different sets of issues (due to bugs in older versions or whatnot).

**Comment by sanmai** [comment]
@michelf Thanks for considering this PR! Please let me know if you need anything else or something isn't clear. Hopefully you can squash-merge this mess of commits.

**Comment by sanmai** [comment]
Build is passing :tada:

**Pull request merged**